shader/antialiasing/cmaa2/process_candidates.comp

```diff
@@ -0,0 +1,349 @@
+#version 460
+
+#extension GL_ARB_separate_shader_objects       : enable
+#extension GL_GOOGLE_include_directive          : enable
+#extension GL_EXT_control_flow_attributes       : enable
+
+#include "cmaa2_common.glsl"
+
+layout(local_size_x = CMAA2_PROCESS_CANDIDATES_NUM_THREADS, local_size_y = 1, local_size_z = 1) in;
+
+// this reschedules final part of work from few to all threads to increase hardware thread occupancy
+#define CMAA2_COLLECT_EXPAND_BLEND_ITEMS 1
+
+#if CMAA2_COLLECT_EXPAND_BLEND_ITEMS
+  const uint CMAA2_BLEND_ITEM_SLM_SIZE = 768;
+  shared uint g_groupSharedBlendItemCount;
+  shared uvec2 g_groupSharedBlendItems[CMAA2_BLEND_ITEM_SLM_SIZE];
+#endif
+
+void StoreColorSample(uvec2 pixelPos, vec3 color, bool isComplexShape, uint msaaSampleIndex) {
+  uint counterIndex = atomicAdd(g_workingControlBuffer[12], 1);
+
+  uvec2 quadPos = pixelPos / uvec2(2, 2);
+  uint offsetXY = (pixelPos.y % 2) * 2 + (pixelPos.x % 2);
+  uint header = (offsetXY << 30) | (msaaSampleIndex << 27) | (uint(isComplexShape) << 26);
+
+  uint counterIndexWithHeader = counterIndex | header;
+
+  uint originalIndex = imageAtomicExchange(g_workingDeferredBlendItemListHeads, ivec2(quadPos), counterIndexWithHeader);
```

Comment:
NOTE: Engine emulates `imageAtomic*` on Metal 2.x and only for `r32ui`.
Should not be an issue in this code, just a note.